processor/intervalprocessor/internal/metrics/metrics.go

```diff
@@ -0,0 +1,23 @@
+// Copyright The OpenTelemetry Authors
+// SPDX-License-Identifier: Apache-2.0
+
+package metrics // import "github.com/open-telemetry/opentelemetry-collector-contrib/processor/intervalprocessor/internal/metrics"
+
+import (
+	"go.opentelemetry.io/collector/pdata/pcommon"
+	"go.opentelemetry.io/collector/pdata/pmetric"
+)
+
+type DataPointSlice[DP DataPoint[DP]] interface {
```

Comment:
these types are only locally used once in `aggregateDataPoints`, doesn't need an extra package, right?

Reply:
Yes, that package has gotten much much smaller since the original design. That said, I'm working on a PR for another component, and needed the same interfaces. So I'm potentially going to put them in `internal/exp/metrics/streams`. See: 

https://github.com/RichieSams/opentelemetry-collector-contrib/commit/d16a21544dbcd3fc52a3870c8c8ac9376a3dca0b#diff-45d5f87d8148e341ebb68b4125ebe4296ccfbb1a9b63a45009358e7a0b617d26

Thoughts?